deploy/basic/terraform/dns.tf

```diff
@@ -0,0 +1,51 @@
+resource "oci_dns_zone" "mushop_dns_zone" {
+  # Conditional
+  count = var.enable_dns_zone ? 1 : 0
```

Comment:
This should also depend on free-tier `var.use_only_always_free_eligible_resources`. WAF is not included in the Always Free tier presently.